test-resources/headless-core-stub/lambdas/callback/src/callback-handler.ts

```diff
@@ -1,12 +1,91 @@
-import type { LambdaInterface } from "@aws-lambda-powertools/commons/types";
+import { LambdaInterface } from "@aws-lambda-powertools/commons/types";
+import { APIGatewayProxyEvent, APIGatewayProxyResult, Context } from "aws-lambda";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { ConfigurationHelper } from "./services/configuration-helper";
+import { CallBackService } from "./services/callback-service";
+import { generatePrivateJwtParams } from "./services/private-key-jwt-helper";
+import { JWK } from "jose";
 
+const sessionTableName = process.env.SESSION_TABLE || "session-common-cri-api";
+const configurationHelper = new ConfigurationHelper();
+const callbackService = new CallBackService();
+const logger = new Logger();
 export class CallbackLambdaHandler implements LambdaInterface {
-    async handler(): Promise<{ status: string; body: string }> {
-        return {
-            status: "200",
-            body: "Hello callback",
-        };
+    async handler(event: APIGatewayProxyEvent, _context: Context): Promise<APIGatewayProxyResult> {
+        try {
+            const authorizationCode = event.queryStringParameters?.authorizationCode as string;
+            logger.info({ message: "Received authorizationCode", authorizationCode });
+
+            const sessionItem = await callbackService.getSessionByAuthorizationCode(
+                sessionTableName,
+                authorizationCode,
+            );
+            logger.info({ message: "Fetching session item...", ...sessionItem });
```

Comment:
`Fetched`

Reply:
Can we make sure the correct verb is used in other places too so we don't get confused on when the action happens - just gives clarity when we debug in the future.